src/stdatamodels/fits_support.py

```diff
@@ -163,6 +164,8 @@ def _make_hdu(hdulist, hdu_name, index=None, hdu_type=None, value=None):
     if hdu_type == fits.PrimaryHDU:
         hdu = hdu_type(value)
     else:
+        if isinstance(value, NDArrayType):
```

Comment:
This check needs to be higher up in the chain of methods.  The `hdu = hdu_type(value)` line just above will also fail if passed a NDArrayType, and there may be problems when updating an existing HDU that the check here doesn't address.

Reply:
Since `hdu_type` is only called in `_make_hdu` I kept the conditional in that function, but moved it to the beginning of the function.